src/Symfony/Component/Intl/Tests/LanguagesTest.php

```diff
@@ -952,40 +1769,30 @@ public function testAlpha3CodeExists()
      */
     public function testGetAlpha3Name($displayLocale)
     {
-        $names = Languages::getNames($displayLocale);
+        $names = Languages::getAlpha3Names($displayLocale);
 
-        foreach ($names as $alpha2 => $name) {
-            $alpha3 = self::$alpha2ToAlpha3[$alpha2] ?? false;
-            if ($alpha3) {
-                $this->assertSame($name, Languages::getAlpha3Name($alpha3, $displayLocale));
-            }
+        foreach ($names as $language => $name) {
+            $this->assertSame($name, Languages::getAlpha3Name($language, $displayLocale));
         }
     }
 
-    public function testGetAlpha3NameWithInvalidCountryCode()
+    public function testGetAlpha3NameWithInvalidLanguageCode()
     {
         $this->expectException(MissingResourceException::class);
 
-        Languages::getAlpha3Name('ZZZ');
+        Languages::getAlpha3Name('zzz');
     }
 
     /**
      * @dataProvider provideLocales
      */
     public function testGetAlpha3Names($displayLocale)
     {
-        $names = Languages::getAlpha3Names($displayLocale);
+        $languages = array_keys(Languages::getAlpha3Names($displayLocale));
 
-        $alpha3Codes = array_keys($names);
-        sort($alpha3Codes);
-        $this->assertSame(array_values(self::$alpha2ToAlpha3), $alpha3Codes);
+        sort($languages);
 
-        $alpha2Names = Languages::getNames($displayLocale);
-        foreach ($alpha2Names as $alpha2Code => $name) {
-            if (!isset(self::$alpha2ToAlpha3[$alpha2Code])) {
-                unset($alpha2Names[$alpha2Code]);
-            }
-        }
-        $this->assertSame(array_values($alpha2Names), array_values($names));
```

Comment:
well the test is not removed, it's now similar to how getNames is tested in `testGetNames`

the point is `getNames` still holds alpha2 codes, and `getAlpha3Names` holds alpha3 codes not in `getNames`. So it's not the same list.

for both lists we validate the expected list of codes already (that's most important IMHO), the labels are more or less an implementation detail IMHO. But sure, this area may be improved by testing against some known fixtures.

Reply:
But `testGetNames` is a weak test (only testing half of the result) because it has nothing to test it against.

But here we have the values from `Languages::getNames` to test against.

If you did that you would dicover f.ex. that `sh` is missing an equivalent value among the alpha3 codes.

Reply:
It is not the same list, but it should be the same list. That is the test you removed.

Why should it not be the same list? I think it is a bug that it is not the same list.

Reply:
see #33165

Reply:
So, you are saying it will be the same list in #33165 ?

Reply:
after both #33148 and #33165 we could verify that yes

Reply:
Ok, that is good then